include/onnxruntime/core/optimizer/graph_transformer_config.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+namespace onnxruntime {
+
+struct GraphTransformerConfiguration {
+  struct PropagateCastOpsConfiguration {
+    // Propagate FP16 Cast operations up and FP32 operations down
+    int level{-1};
+    // Cast propagation strategy. One strategy is to insert casts around all the nodes with the allowed opcodes
+    // and reduce, remove redundent and back-to-back casts, and
+    // the other is to propagate casts using flood-fill approach, expanding float16 regions in the graph
+    // traversinvg the graph up/down.
+    enum class Strategy {
+      InsertAndReduce = 0,
+      FloodFill = 1
+    };
+    Strategy strategy = Strategy::InsertAndReduce;
+    // List of allowed opcodes to consider as safe to execute in float16, while moving cast operations
+    std::vector<std::string> allow;
+    // Whether allow fusion of layer norm subgraph if doing so will cause modified precision.
+  };  // Whether to enable GELU approximation which is faster but produces different results.
+  PropagateCastOpsConfiguration propagate_cast_ops_config;
```

Comment:
I am wondering if this will be used in the InferenceSession... Is PropagateCastOpsConfiguration going to be useful for inference case? 

If not, we should probably place it in the TrainingGraphTransformerConfiguration... and it will be no need for the base class GraphTransformerConfiguration

Reply:
GraphTransformerConfiguration may not be used in the InferenceSession, at least for now. However the reason I had to create this class hierarchy is the namespace.  PropagateCastOps is in the onnxruntime name space. If I create a class under orttraining/orttraining then it belongs to orttraining namespace. I thought that creating a dependence in onnxruntime namespace on something define in the orttraining namespace.